### DISCUSSION TITLE
No HEIC converter found

### DISCUSSION THREAD

**Discussion by iswaterbad** [ReagentX/imessage-exporter#300] [Q&A] [answered]
I’m trying to extract all the images from an iMessage backup, but it says “no heic converter found” when extracting the messages. How can I fix this?

Also, is it even possible to get all the images extracted into a folder? (Could be heic too, I just assumed this was part of the issue, since I’m not seeing any images on either html or txt)

edit: I’m on windows

**Comment by ReagentX** [comment]
What command are you running?

**Reply by iswaterbad** [reply]
`$ imessage-exporter -f txt -p ~/iphone_backup_latest -a iOS -o backup_export `
This one from the readme, and I changed the ~/iphone_backup_latest to my backup location. I also tried adding -c compatible, but had the same result.

**Reply by ReagentX** [reply, marked as answer]
The issue with `-c compatible` and iOS backups was fixed in #296 and will be released in California Poppy (1.10). 

You are getting the HEIC converter message because you do not have one installed on your machine. See #258 for details.

**Reply by iswaterbad** [reply]
Ok thank you! I’ll try it when I get back to my computer.

**Reply by iswaterbad** [reply]
Ok, so now it does actually make an attachments folder (even though it still says 'no heic converter found'). This is perfectly fine for me, as i just wanted the photos, and might even be related to #296. 
The other thing is that I'm getting a few "attachment not found at specified path" errors. This could be deleted photos or something, but not that many that it matter much to me.

Anyways, thank you very much for this app and for the help. It did just what I needed, and I love that there is an open source alternative to the other apps.

**Reply by ReagentX** [reply]
If you run with `-d`, `imessage-exporter` will output some diagnostics about attachments (and other things). A detailed explanation of the output is located [here](https://github.com/ReagentX/imessage-exporter/blob/develop/docs/diagnostics.md).

**Reply by iswaterbad** [reply]
Ok I ran the command and got 4% missing files. Overall not a large part of the data, plus after reading the explanation I’m going to assume it’s attachments I’ve already deleted.
